src/app/components/SubmitBar/index.tsx

```diff
@@ -229,10 +256,65 @@ export class SubmitBar extends React.Component<
             closeOptions={this.closeRunOptions}
           />
         ) : null}
+        {isStoryModeModalOpen ? this.storyModeModalComponent(false, current_stars) : null}
       </div>
     );
   }
 
+  public componentDidMount() {
+    this.props.getQuestStatus();
+    this.props.setCurrentLevel(1, 0);
+  }
+
+  private questLevel() {
+    const levelsComponent = [];
+    if (this.props.ratings) {
+      for (let i = 0; i < levels.length; i += 1) {
+        levelsComponent[i] =
+          this.props.ratings[i] !== undefined ? (
+            <DropDownItem
+              level={String(this.props.ratings[i].level)}
+              rating={this.props.ratings[i].stars}
+              openStoryModeModal={this.props.openStoryModeModal}
+              setCurrentLevel={this.props.setCurrentLevel}
+            />
+          ) : (
+            <DropDownItem
+              level={String(i + 1)}
+              rating={-1}
+              openStoryModeModal={this.props.openStoryModeModal}
+              setCurrentLevel={this.props.setCurrentLevel}
+            />
+          );
+      }
+      return levelsComponent;
+    }
+    return null;
```

Comment:
Use destructuring. How many times you need to type ```this.props``` before you realize there must be something better?